baseplate/clients/requests.py

```diff
@@ -0,0 +1,322 @@
+        self.span = span
+
+    def delete(self, url: str, **kwargs: Any) -> Response:
+        """Send a DELETE request.
+
+        See :py:func:`requests.request` for valid keyword arguments.
+
+        """
+        return self.request("DELETE", url, **kwargs)
+
+    def get(self, url: str, **kwargs: Any) -> Response:
+        """Send a GET request.
+
+        See :py:func:`requests.request` for valid keyword arguments.
+
+        """
+        return self.request("GET", url, **kwargs)
+
+    def head(self, url: str, **kwargs: Any) -> Response:
+        """Send a HEAD request.
+
+        See :py:func:`requests.request` for valid keyword arguments.
+
+        """
+        return self.request("HEAD", url, **kwargs)
+
+    def options(self, url: str, **kwargs: Any) -> Response:
+        """Send an OPTIONS request.
+
+        See :py:func:`requests.request` for valid keyword arguments.
+
+        """
+        return self.request("OPTIONS", url, **kwargs)
+
+    def patch(self, url: str, **kwargs: Any) -> Response:
+        """Send a PATCH request.
+
+        See :py:func:`requests.request` for valid keyword arguments.
+
+        """
+        return self.request("PATCH", url, **kwargs)
+
+    def put(self, url: str, **kwargs: Any) -> Response:
+        """Send a PUT request.
+
+        See :py:func:`requests.request` for valid keyword arguments.
+
+        """
+        return self.request("PUT", url, **kwargs)
+
+    def prepare_request(self, request: Request) -> PreparedRequest:
+        """Construct a :py:class:`~requests.PreparedRequest` for later use.
+
+        The prepared request can be stored or manipulated and then used with
+        :py:meth:`send`.
+
+        """
+        return request.prepare()
+
+    def request(self, method: str, url: Union[str, bytes], **kwargs: Any) -> Response:
+        """Send a request.
+
+        :param method: The HTTP method of the request, e.g. ``GET``, ``PUT``, etc.
+        :param url: The URL to send the request to.
+
+        See :py:func:`requests.request` for valid keyword arguments.
+
+        """
+        request = Request(method=method.upper(), url=url, **kwargs)
+        prepared = self.prepare_request(request)
+        return self.send(prepared)
+
+    def send(self, request: PreparedRequest) -> Response:
+        """Send a :py:class:`~requests.PreparedRequest`."""
+
+        with self.span.make_child(f"{self.name}.request") as span:
+            span.set_tag("http.method", request.method)
+            span.set_tag("http.url", request.url)
+
+            # we cannot re-use the same session every time because sessions re-use the same
+            # CookieJar and so we'd muddle cookies cross-request. if the application wants
+            # to keep track of cookies, it should do so itself.
+            #
+            # note: we're still getting connection pooling because we're re-using the adapter.
+            session = Session()
+            session.mount("http://", self.adapter)
```

Comment:
:no_bell: both of these are needed so it can support sending `http://` requests?

Reply:
yup!